src/posix/platform/netif.cpp

```diff
@@ -1080,6 +1080,39 @@ static otError tryProcessIcmp6RaMessage(otInstance *aInstance, const uint8_t *da
 }
 #endif // OPENTHREAD_CONFIG_BORDER_ROUTING_DHCP6_PD_ENABLE
 
+/**
+ * Returns whether the address is a required anycast address (RFC2373, 2.6.1).
+ *
+ */
+static bool isRequiredAnycast(const uint8_t *aAddress, uint8_t aPrefixLength)
+{
+    static constexpr uint16_t kIp6AddressSize   = 16;
+    bool                      isRequiredAnycast = true;
+    int                       firstBytePos      = aPrefixLength / 8;
+
+    if (aPrefixLength % 8 > 0)
+    {
+        firstBytePos++;
+    }
+
+    // Check the last byte of the prefix and the remaining bits
+    if ((aAddress[firstBytePos] & ((1 << (aPrefixLength % 8)) - 1)) != 0)
+    {
+        isRequiredAnycast = false;
+    }
```

Comment:
Can consider moving this check under `if (aPrefixLanegth % 8 > 0)` block (when we need to check partial bits in last prerix byte). If the `aPrefixLength` is a multiple of 8, then we dont need to do any bitmask check (the code as is would compare it with a `0` mask which is fine, but not necessary). I think this makes the code easier the read. With this, the `for` loop can start from `Pos` and not `Pos + 1`.
```suggestion
    uint8_t remainingBits = aPrefixLength % 8;

    if (remainingBits != 0)
    {
        if ((aAddress[firstBytePos] & ((1 << remainingBits) - 1)) != 0)
        {
            isRequiredAnycast = false;
        }

        firstBytePos++;
    }

    for (int i = firstBytePos; i < kIp6AddressSize; ++i)
    {
        if (aAddress[i] != 0)
        {
            isRequiredAnycast = false;
            break;
        }
    }
```